backend/battle/forms.py

```diff
@@ -53,6 +58,10 @@ def clean(self):
         pokemon_3 = self.cleaned_data.get('pokemon_3')
         obj_battle = cleaned_data['battle']
         obj_trainer = cleaned_data['trainer']
+        trainer_on_database = User.objects.filter(email=obj_trainer)
```

Comment:
Why do you need those lines? `obj_trainer` should already be a reference to the trainer in the DB.